persistence/jdbc-common/src/main/java/org/infinispan/persistence/jdbc/common/impl/connectionfactory/ManagedConnectionFactory.java

```diff
@@ -97,10 +70,43 @@ public Connection getConnection() throws PersistenceException {
    @Override
    public void releaseConnection(Connection conn) {
       try {
-         if (conn != null) // Could be null if getConnection failed
+         if (conn != null) {
+            // Could be null if getConnection failed
             conn.close();
+         }
       } catch (SQLException e) {
          PERSISTENCE.sqlFailureClosingConnection(conn, e);
       }
    }
+
+   private void initDataSource() {
+      dataSource = managedConfiguration.dataSource();
+      if (dataSource != null) {
+         return;
+      }
+      InitialContext ctx = null;
+      String datasourceName = managedConfiguration.jndiUrl();
+      try {
+         ctx = new InitialContext();
+         DataSource dataSource = (DataSource) ctx.lookup(datasourceName);
+         if (log.isTraceEnabled()) {
+            log.tracef("Datasource lookup for %s succeeded: %b", datasourceName, dataSource);
+         }
+         if (dataSource == null) {
+            PERSISTENCE.connectionNotFound("JNDI", datasourceName);
```

Comment:
Not related to your PR, but you can throw the exception created by the logging framework.
`throw PERSISTENCE.connectionNotFound("JNDI", datasourceName);`